src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.js

```diff
@@ -450,7 +463,7 @@ export const AzureComputeModalBase = ({
       ],
       [Utils.DEFAULT, renderMainForm]
     ),
-    loading && spinnerOverlay,
+    (loading || isLoadingCloudEnvironments) && spinnerOverlay,
```

Comment:
perhaps just rename the original `loading` to be something more representative of the fact its modal-specific state.  Something like `modalRefreshing`